java/flight/src/main/java/org/apache/arrow/flight/FlightServer.java

```diff
@@ -247,6 +254,9 @@ public FlightServer build() {
         return null;
       });
 
+      if (!interceptors.isEmpty()) {
```

Comment:
same question as above, is it really that much more efficient to avoid always creating?